pygmt/base_plotting.py

```diff
@@ -849,70 +858,128 @@ def text(
-                if not os.path.exists(textfile):
-                    raise GMTInvalidInput(f"Cannot find the file: {textfile}")
+        # Ensure inputs are either textfiles, x/y/text, or position/text
+        if position is None:
+            kind = data_kind(textfiles, x, y, text)
+        elif position is not None:
+            if x is not None or y is not None:
+                raise GMTInvalidInput(
+                    "Provide either position only, or x/y pairs, not both"
+                )
+            kind = "vectors"
 
-        if angle is not None or font is not None or justify is not None:
+        if kind == "vectors" and text is None:
+            raise GMTInvalidInput("Must provide text with x/y pairs or position")
+
+        # Build the `-F` argument in gmt text.
+        if (
+            position is not None
+            or angle is not None
+            or font is not None
+            or justify is not None
+        ):
             if "F" not in kwargs.keys():
                 kwargs.update({"F": ""})
-            if angle is not None and isinstance(angle, (int, float)):
+            if angle is not None and isinstance(angle, (int, float, str)):
                 kwargs["F"] += f"+a{str(angle)}"
             if font is not None and isinstance(font, str):
                 kwargs["F"] += f"+f{font}"
             if justify is not None and isinstance(justify, str):
                 kwargs["F"] += f"+j{justify}"
+            if position is not None and isinstance(position, str):
+                kwargs["F"] += f'+c{position}+t"{text}"'
 
         with GMTTempFile(suffix=".txt") as tmpfile:
             with Session() as lib:
                 if kind == "file":
                     fname = textfiles
                 elif kind == "vectors":
-                    pd.DataFrame.from_dict(
-                        {
-                            "x": np.atleast_1d(x),
-                            "y": np.atleast_1d(y),
-                            "text": np.atleast_1d(text),
-                        }
-                    ).to_csv(
-                        tmpfile.name,
-                        sep="\t",
-                        header=False,
-                        index=False,
-                        quoting=csv.QUOTE_NONE,
-                    )
+                    if position is not None:
+                        fname = ""
+                    else:
+                        pd.DataFrame.from_dict(
+                            {
+                                "x": np.atleast_1d(x),
+                                "y": np.atleast_1d(y),
+                                "text": np.atleast_1d(text),
+                            }
+                        ).to_csv(
+                            tmpfile.name,
+                            sep="\t",
+                            header=False,
+                            index=False,
+                            quoting=csv.QUOTE_NONE,
+                        )
```

Comment:
```suggestion
                        file_context = lib.virtualfile_from_vectors(
                            np.atleast_1d(x), np.atleast_1d(y), np.atleast_1d(text)
                        )
```

@leouieda mentioned before at https://github.com/GenericMappingTools/pygmt/pull/321#discussion_r334993927 to use virtualfiles, and I finally figured out what he meant. Unfortunately, `virtualfile_from_vectors` doesn't yet support `str` types. This is the error on `_check_dtype_and_dim`:

```python-traceback
        if array.dtype.name not in DTYPES:
            raise GMTInvalidInput(
>               "Unsupported numpy data type '{}'.".format(array.dtype.name)
            )
E           pygmt.exceptions.GMTInvalidInput: Unsupported numpy data type 'str864'.

```

From what I can tell, we'll need to use [`GMT_Put_Strings`](https://docs.generic-mapping-tools.org/latest/api.html?highlight=gmt_char#gmt-put-strings) is that right? We should do this refactor in a separate PR, this one is already getting a bit too long.

Reply:
Yes, as I understand it, we need to use `GMT_Put_Strings` to pass trailing strings to GMT.